packages/block-editor/src/components/list-view/block-contents.js

```diff
@@ -61,23 +65,96 @@ const ListViewBlockContents = forwardRef(
 			? selectedClientIds
 			: [ clientId ];
 
+		const { blockName, blockAttributes } = useSelect(
+			( select ) => {
+				const blockObject =
+					select( blockEditorStore ).getBlock( clientId );
+				return {
+					blockName: blockObject?.name,
+					blockAttributes: blockObject?.attributes,
+				};
+			},
+			[ clientId ]
+		);
+
+		const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+		const metaDataSupport = getBlockSupport(
+			blockName,
+			'__experimentalMetadata',
+			false
+		);
+
+		const supportsBlockNaming = !! (
+			true === metaDataSupport || metaDataSupport?.name
+		);
+
+		const { isLocked } = useBlockLock( clientId );
+
+		const toggleLabelEditingMode = ( value ) => {
+			if ( ! supportsBlockNaming ) {
+				return;
+			}
+
+			setLabelEditingMode( value );
+		};
+
+		const blockTitle = useBlockDisplayTitle( {
+			clientId,
+			context: 'list-view',
+		} );
+
+		const blockInformation = useBlockDisplayInformation( clientId );
+
+		const className = classnames( 'block-editor-list-view-block-contents', {
+			'is-dropping-before': isBlockMoveTarget,
+			'has-block-naming-support': supportsBlockNaming,
+		} );
+
+		function inputSubmitHandler( updatedInputValue ) {
+			updateBlockAttributes( clientId, {
+				// Include existing metadata (if present) to avoid overwriting existing.
+				metadata: {
+					...( blockAttributes?.metadata &&
+						blockAttributes?.metadata ),
+					name: updatedInputValue,
+				},
+			} );
+		}
+
+		if ( labelEditingMode ) {
+			return (
+				<ListViewBlockInput
+					ref={ ref }
+					className={ className }
+					onToggleExpanded={ onToggleExpanded }
+					toggleLabelEditingMode={ toggleLabelEditingMode }
+					blockInformation={ blockInformation }
+					isLocked={ isLocked }
+					blockTitle={ blockTitle }
+					onSubmit={ inputSubmitHandler }
+					{ ...props }
+				/>
+			);
+		}
+
 		return (
 			<BlockDraggable clientIds={ draggableClientIds }>
 				{ ( { draggable, onDragStart, onDragEnd } ) => (
 					<ListViewBlockSelectButton
 						ref={ ref }
 						className={ className }
-						block={ block }
 						onClick={ onClick }
 						onToggleExpanded={ onToggleExpanded }
-						isSelected={ isSelected }
-						position={ position }
-						siblingBlockCount={ siblingBlockCount }
-						level={ level }
-						draggable={ draggable }
```

Comment:
were these things passed down for no reason? the diff is unclear but it looks like that ...